docs/messaging/kafka.md

```diff
@@ -44,6 +49,62 @@ If an intermediary broker is present, `service.name` and `peer.service` will not
 
 `messaging.client_id` SHOULD be set to the `client-id` of consumers, or to the `client.id` property of producers.
 
+## Metrics
+
+### Metric: `messaging.publish.duration`
+
+Measures time it takes to publish message or a batch of messages to Apache Kafka including all retries.
+
+This metric is [required][MetricRequired]
+
+This metric follows the common [messaging.publish.duration](./messaging-metrics.md#metric-messagingpublishduration) definition.
+
+**Notes:**
+
+- The following attributes (with corresponding [requirement levels](#attributes)) are defined in addition to generic attributes for this metric:
+  * `messaging.kafka.destination.partition`
+
+### Metric: `messaging.publish.messages`
+
+Measures the number of published messages.
+
+This metric is [required][MetricRequired]
+
+This metric follows the common [messaging.publish.messages](./messaging-metrics.md#metric-messagingpublishmessages) definition.
+
+**Notes:**
+
+- The following attributes (with corresponding [requirement levels](#attributes)) are defined in addition to generic attributes for this metric:
+  * `messaging.kafka.destination.partition`
+
+### Metric: `messaging.receive.duration`
+
+Measures duration of consumer client call that polls messages.
+
+This metric is [required][MetricRequired].
+
+This metric follows the common [messaging.receive.duration](./messaging-metrics.md#metric-messagingdeliverduration) definition.
+
+**Notes:**
+
+- The following attributes (with corresponding [requirement levels](#attributes)) are defined in addition to generic attributes for this metric:
+  * `messaging.kafka.destination.partition`
+  * `messaging.kafka.consumer.group`
+
+### Metric: `messaging.receive.messages`
+
+Measures the number of received messages.
+
+This metric is [required][MetricRequired].
+
+This metric follows the common [messaging.receive.messages](./messaging-metrics.md#metric-messagingreceivemessages) definition.
+
+**Notes:**
+
+- The following attributes (with corresponding [requirement levels](#attributes)) are defined in addition to generic attributes for this metric:
+  * `messaging.kafka.destination.partition`
+  * `messaging.kafka.consumer.group`
```

Comment:
We will need to clarify with the wider semantic conventions community whether this is the way we want to extend generic metric definitions.

The approach proposed in this PR is endorsed by the messaging workgroup, because it allows telemetry consumers to utilize a single metric (for example `messaging.receive.messages`) to measure throughput, regardless of the messaging system used. This makes it much easier to build generic APM experiences.

For obtaining system-specific information, it is still possible to utilize the additional system-specific attributes on the metric.